lib/goth/token.ex

```diff
@@ -112,9 +120,14 @@ defmodule Goth.Token do
     JOSE.JWT.sign(jwk, header, claim_set) |> JOSE.JWS.compact() |> elem(1)
   end
 
-  defp request(http_client, url, {:instance, account}) do
+  defp maybe_override_token_scope({:metadata, _, scope}, _), do: scope
+  defp maybe_override_token_scope(_, scope), do: scope
+
+  defp request(http_client, url, {:metadata, account, scope}) do
     headers = [{"metadata-flavor", "Google"}]
-    url = "#{url}/computeMetadata/v1/instance/#{account}/token"
+    base_url = if(url == @default_url, do: @metadata_url, else: url)
```

Comment:
these checks feel a bit dirty. It might be because the way we were already handling the defaults, perhaps we were injecting the default values too "soon". Another idea would be, unless the user explicitly set these options, to keep them as `nil` as long as possible and only at the very last moment fallback that nil to a proper default value if that would make sense in that scenario. I really want to avoid the situation where it's unclear wether the user happened to explicitly set the value that is equal to the default or they didn't enter it at all. Then, if the user is setting the scope & instance where they shouldn't, we'd respect their choice and let it error out on the google side of things so they could correct their configuration. WDYT? Even if this would sound like a reasonable idea we don't have to worry about it now, I'd be happy to look into it after this lands.